extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -818,31 +905,21 @@ class LifecycleTask extends BackbeatTask {
      * @return {undefined}
      */
     _applyTransitionRule(params, log) {
-        const entry = ActionQueueEntry.create('copyLocation')
-              .setResultsTopic(this.objectTasksTopic)
-              .addContext({
-                  origin: 'lifecycle',
-                  ruleType: 'transition',
-                  reqId: log.getSerializedUids(),
-              })
-              .setAttribute('target.bucket', params.bucket)
-              .setAttribute('target.key', params.objectKey)
-              .setAttribute('target.version', params.encodedVersionId)
-              .setAttribute('target.eTag', params.eTag)
-              .setAttribute('target.lastModified', params.lastModified)
-              .setAttribute('toLocation', params.site);
-
-        this._sendDataMoverAction(entry, err => {
+        async.waterfall([
+            next =>
+                this._getObjectMD(params, log, next),
```

Comment:
Can't we instead replace the `headObject` done earlier by this call? It sounds wasteful to do two metadata fetches per object (even three if we count the cloud one).

Reply:
As discussed, it looks like we can remove the `headObject` request altogether but requires further investigation wether we should instead just replace with the object MD request. Will follow with a separate PR for this specific change because it pertains to the expiration feature code path.